File: app/models/playing_card.rb
require_relative 'card_deck'
require_relative 'go_fish'

class PlayingCard
  attr_reader :rank, :suit

  def initialize(rank, suit)
    @rank = rank
    @suit = suit
  end

  def value
    return 11 if rank == "Jack"
    return 12 if rank == "Queen"
    return 13 if rank == "King"
    return 14 if rank == "Ace"
    rank.to_i
  end

  def image_name
    rank == "10" ? "#{rank}#{suit[0]}" : "#{rank[0]}#{suit[0]}"
  end

  def to_s
    "/images/card_faces/#{image_name}"
  end

  def alt_to_s
    "#{rank} of #{suit}"
  end

  def ==(other)
    rank == other&.rank && suit == other&.suit
  end
end

File: app/javascript/models/PlayingCard.js
class PlayingCard {
  constructor(rank, suit) {
    this._rank = rank
    this._suit = suit
  }

  rank() {
    return this._rank
  }

  suit() {
    return this._suit
  }

  value() {
    if (this._rank[0] === "J") {
      return 11
    } else if (this._rank[0] === "Q") {
      return 12
    } else if (this._rank[0] === "K") {
      return 13
    } else if (this._rank[0] === "A") {
      return 14
    } else {
      return parseInt(this.rank())
    }
  }

  to_s() {
    return `${this.rank()} of ${this.suit()}`
  }

  cssId() {
    return `${this.rank()}-${this.suit()}`
  }

  imageName() {
    if (this.rank() == "10") {
      return `${this.rank()}${this.suit()[0]}.jpg`
    } else {
      return `${this.rank()[0]}${this.suit()[0]}.jpg`
    }
  }
}

const imageHash = {'8C': require('../images/card_faces/8C.jpg')}

export default PlayingCard
export { imageHash }
